client/signup/main.jsx

```diff
@@ -224,6 +228,12 @@ class Signup extends React.Component {
 
 	handleSignupFlowControllerCompletion = ( dependencies, destination ) => {
 		const filteredDestination = getDestination( destination, dependencies, this.props.flowName );
+
+		// If the filtered destination is different from the flow destination, then save it in the store
```

Comment:
```suggestion
		// If the filtered destination is different from the flow destination (e.g. changes to checkout), then save the flow destination so the user ultimately arrives there
```

Would have saved me a few clicks trying to figure out what `getDestination` does. Alternatively we could rename `getDestination` to something else?